examples/external-dns/main.tf

```diff
@@ -0,0 +1,97 @@
+## locals
+
+locals {
+  plan_id = data.duplocloud_tenant.current.plan_id
+}
+
+## Datasources
+
+# Get tenant info, typically from the current workspace
+data "duplocloud_tenant" "current" {
+  name = var.tenant_name
+}
+
+# Get EKS credentials for helm provider
+data "duplocloud_eks_credentials" "current" {
+  plan_id = local.plan_id
+}
+
+# Get infrastructure to get the region for aws provider
+data "duplocloud_infrastructure" "current" {
+  infra_name = local.plan_id
+}
+
+# Get the route53 zone information
+data "aws_route53_zone" "selected" {
+  name = var.public_dns_domain
+}
+
+## Provider versions
+
+terraform {
+  required_version = ">= 1.4.4"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.12.0"
+    }
+    duplocloud = {
+      source  = "duplocloud/duplocloud"
+      version = "~> 0.10.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = "~> 2.11.0"
+    }
+  }
+  backend "s3" {
+    workspace_key_prefix = "duplocloud/addons"
+    key                  = "external-dns"
+    encrypt              = true
+  }
+}
+
+## Provider configs
+
+# Configure helm provider for chart deployment
+provider "helm" {
+  kubernetes {
+    host                   = data.duplocloud_eks_credentials.current.endpoint
+    cluster_ca_certificate = data.duplocloud_eks_credentials.current.ca_certificate_data
+    token                  = data.duplocloud_eks_credentials.current.token
+  }
+}
+
+# Configure aws provider for IAM role extension
+provider "aws" {
+  region = data.duplocloud_infrastructure.current.region
+}
+
+## Modules and resources
+
+# Deploy external-dns module for managing Route53 records.  
+# See https://github.com/kubernetes-sigs/external-dns/tree/master/docs/tutorials
+module "external_dns" {
+  source           = "../../modules/external-dns"
+  tenant_name      = var.tenant_name
+  dns_provider     = "aws"
+  route53_zone_arn = data.aws_route53_zone.selected.arn
+  values = {
+    "aws.zone" : "public",
+    "txtOwnerId" : data.aws_route53_zone.selected.id
```

Comment:
About `txtOwnerId`. This is actually just a unique name to determine records owned by this instance. This should be a human readable name. Using the zone id would conflict if two instances were using the same zone, I've accidentally done this before. Can you determine the name of the cluster/plan based on the the given tenant and use cluster name instead? It would be very rare for a customer to want to manually set this value.

Reply:
I updated this.